### ISSUE TITLE
string_test is abandoned

### ISSUE THREAD

**Issue by mumbleskates** [fastfloat/fast_float#106]
OS: ubuntu 21.04
compiler: clang-14 nightly

It seems that some of the extended tests are not run regularly. I cannot get string_test to pass, and it's unclear when if ever this test is run by CI.

Building and running with the built-in cmake:
```
Test project /home/widders/repos/fast_float/build
    Start 1: example_test
1/4 Test #1: example_test .....................   Passed    0.00 sec
    Start 2: example_comma_test
2/4 Test #2: example_comma_test ...............   Passed    0.00 sec
    Start 3: basictest
3/4 Test #3: basictest ........................   Passed    0.00 sec
    Start 4: string_test
4/4 Test #4: string_test ......................Child aborted***Exception:   0.00 sec
32 bits checks
parsing  1e1000 100000 3.14159265359  -1e-500 001    1e01  1e0000001  -inf
 I could not parse 
terminate called after throwing an instance of 'std::runtime_error'
  what():  bug
```

**Comment by lemire** [comment]
Thank you. I will review.

**Comment by lemire** [comment]
Oh. It is meant to fail. Let me update the test.

(This is almost surely not a bug.)

**Comment by lemire** [comment]
Fixed in release 3.1.

The reason that small test failed is that to better emulate the standard, we do not skip white spaces. We used to do so.

**Comment by mumbleskates** [comment]
> The reason that small test failed is that to better emulate the standard, we do not skip white spaces. We used to do so.

Thanks! I figured this was probably why, it looked like it was breaking with a leading space.
